dll/win32/shell32/CDefView.cpp

```diff
@@ -767,40 +833,239 @@ void CDefView::UpdateListColors()
 */
 BOOL CDefView::InitList()
 {
-    SHELLDETAILS sd;
-    WCHAR szTemp[50];
     HIMAGELIST big_icons, small_icons;
 
     TRACE("%p\n", this);
 
     m_ListView.DeleteAllItems();
 
+    Shell_GetImageLists(&big_icons, &small_icons);
+    m_ListView.SetImageList(big_icons, LVSIL_NORMAL);
+    m_ListView.SetImageList(small_icons, LVSIL_SMALL);
+
     m_hMenuArrangeModes = CreateMenu();
+    LoadColumns();
+    return TRUE;
+}
+
+/**********************************************************
+* Column handling
+*/
+static HRESULT SHGetLVColumnSubItem(HWND List, UINT Col)
```

Comment:
Since I would expect that this function would mostly succeed, and return actual indices instead of an error, I would suggest you make it have `INT` return type (or whatever the `LVCOLUMN::iSubItem` type is), instead of "HRESULT" (which means "an error code").
And then, just cast the `E_FAIL` to `INT` or whatever.

Or, make the function return an `UINT` and in case of error, make it return `UINT_MAX`, like what has been done in e.g. the ReallyGetMenuItemID function.

Reply:
Hresult is an int.